### ISSUE TITLE
--metadata-file only contains image digest

### ISSUE THREAD

**Issue by miki725** [docker/buildx#1594]
As per https://github.com/docker/buildx/blob/master/docs/reference/buildx_build.md it seems metadata file should include information about the build, configuration/etc however when I try to use it I only get the container image digest without any additional metadata:

```sh
➜ cat Dockerfile.test
FROM alpine
RUN ls

➜ docker buildx build --load --metadata-file metadata.json -f Dockerfile.test .
[+] Building 0.0s (6/6) FINISHED
 => [internal] load build definition from Dockerfile.test                                                                                                                                                                                                  0.0s
 => => transferring dockerfile: 36B                                                                                                                                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                          0.0s
 => => transferring context: 34B                                                                                                                                                                                                                           0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                                                                                                                                                                           0.0s
 => [1/2] FROM docker.io/library/alpine                                                                                                                                                                                                                    0.0s
 => CACHED [2/2] RUN ls                                                                                                                                                                                                                                    0.0s
 => exporting to image                                                                                                                                                                                                                                     0.0s
 => => exporting layers                                                                                                                                                                                                                                    0.0s
 => => writing image sha256:0a95cd5812b34cfcfa1c7e00a8ac9ad3929ffb2621364cc9fb82456a82c07181                                                                                                                                                               0.0s

➜ cat metadata.json
{
  "containerimage.digest": "sha256:0a95cd5812b34cfcfa1c7e00a8ac9ad3929ffb2621364cc9fb82456a82c07181"
}
```

Are there additional necessary flags which will include rest of the metadata in the metadata file? Not sure how to replicate example from the docs.

**Comment by crazy-max** [comment]
Can you show the output of `docker buildx ls` and `docker info` please?

**Comment by miki725** [comment]
sure sorry about that. meant to paste versions of stuff:

```
➜ uname -a
Linux earth 6.1.4-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 07 Jan 2023 15:10:07 +0000 x86_64 GNU/Linux
```

```
➜ cat /etc/*release
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo
```

```
➜ docker info
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., 0.9.1)
  compose: Docker Compose (Docker Inc., 2.14.2)

Server:
 Containers: 4
  Running: 4
  Paused: 0
  Stopped: 0
 Images: 43
 Server Version: 20.10.22
 Storage Driver: btrfs
  Build Version: Btrfs v6.0.2
  Library Version: 102
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 9ba4b250366a5ddde94bb7c9d1def331423aa323.m
 runc version:
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 6.1.4-arch1-1
 Operating System: Arch Linux
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 62.52GiB
 Name: earth
 ID: 5223:HSTA:MFNN:ROPL:BHNY:BFUN:6A5D:6MQA:PCKQ:YMYK:BI5H:TX6C
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

```
➜ docker buildx version
github.com/docker/buildx 0.9.1 ed00243a0ce2a0aee75311b06e32d33b44729689
```

```
➜ docker buildx ls
NAME/NODE    DRIVER/ENDPOINT STATUS  BUILDKIT PLATFORMS
default *    docker
  default    default         running 20.10.22 linux/amd64, linux/386
node-amd64   docker
  node-amd64 node-amd64      running 20.10.22 linux/amd64, linux/386
node-arm64   docker
  node-arm64 node-arm64      running 20.10.22 linux/amd64, linux/386
```

**Comment by crazy-max** [comment]
Docker `20.10.22` (backed by BuildKit 0.8.2) does not have the buildinfo feature: https://github.com/moby/buildkit/blob/master/docs/build-repro.md#metadata-json-output. This is available since BuildKit 0.10 (Docker 23). Also keep in mind that buildinfo is deprecated in BuildKit 0.11 and has been replaced by the provenance attestation feature. More info: https://docs.docker.com/build/attestations/slsa-provenance/
